add a Vertx wrapper for the MappedBatchLoaderWithContext

craig-day opened the pull request:
This follows the pattern started by wrapping the BatchLoaderWithContext in a
Vertx callback friendly API. Instead of receiving a list of keys and returning
a list of values, this will accept a set of keys and return a map of key-value
pairs. This can be useful for maintaining integrity when a 1-1 mapping can't
be guaranteed, or when parial loads are expected.

https://github.com/graphql-java/java-dataloader#returning-a-map-of-results-from-your-batch-loader

@tsegismont you seem to be active with the graphql package. 
@pmlopes you seem to be all over this project. 

Sorry if this isn't the desired contribution path, but I didn't see a contributing doc or mention of it in the readme. Happy to make adjustments as desired.

tsegismont commented on vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/dataloader/impl/VertxMappedBatchLoaderImpl.java:
```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
```
```suggestion
 * Copyright 2020 Red Hat, Inc.
```

tsegismont commented on vertx-web-graphql/src/main/java/io/vertx/ext/web/handler/graphql/dataloader/VertxMappedBatchLoader.java:
```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2019 Red Hat, Inc.
```
```suggestion
 * Copyright 2020 Red Hat, Inc.
```

craig-day commented:
@tsegismont thanks for the review! I updated the docs and added a test for the `VertxMappedBatchLoader`. 

I also re-used that test and explicitly added a test for the existing `VertxBatchLoader` since I was working on that setup anyway. That is its own commit so I can easily drop it from this PR and open a new PR if you would prefer.

craig-day commented:
I'm not sure why the tests failed, errored out on the mavin install 🤷‍♂

craig-day commented:
FWIW, I can run the test suite locally and it all passes. I'm not sure how to make travis happy. Maybe a repo maintainer can restart the build?

tsegismont commented:
Thanks for the updates @craig-day 

It seems you reused the code from `io.vertx.ext.web.handler.graphql.BatchLoaderTest`. Perhaps it can be deleted now, can't it?

craig-day commented:
perhaps. I guess the slight difference is that the `BatchLoaderTest` just uses a lambda to do the fetching, but the two tests I wrong explicitly instantiate each `Vertx*BatchLoader`. I guess that might make the generic test obsolete. I'll remove it.
